teps/0131-tekton-conformance-policy.md

```diff
@@ -0,0 +1,251 @@
+1. **The Conformance API**: we promote a field from `OPTIONAL` to `REQUIRED`.
+1. **The Policy Itself**: for example,  in the future, we may want to define several conformance levels that vendors can choose to conform to, or we may change the definition of the conformance.
+
+The versioning format: `<Tekton Primitive Type>-<GA API Version>.<Policy / API  Update>`
+* **pipelines.v1.0**: the initial version of **Tekton Pipelines V1 Conformance Policy**.
+* **pipelines.v2.1**: added a new required field in **Tekton Pipelines V2 Conformance APIs**.
+* **results.v2.0**: the initial version of **Tekton Results V2 Conformance Policy**.
+* **results.v2.1**: some policy change on **Tekton Results V2 Conformance Policy**.
+
+Worth noting that even if we bump up the policy version, vendors can still claim the conformance on a previous version.
+
+### Policy Update Procedure
+* **Open a PR** to propose the update.
+* **The PR** must be approved by more than half of [the project OWNERS][project owners] (i.e. 50% + 1) when it involves actual requirements changes (as opposed to typo/grammar fixing).
+* **Update Notice** ahead of time.
+  * Nice to have: conformance test kit can pop up the update notice when vendors run it.
+  * How long do we want the update notice to be ahead of is an open question.
+
+### Conformance Test Suites
+
+The general idea is we follow "[duck test][duck test]", we provide inputs (YAMLs including all required fields) and outputs (status, results etc.) to see if vendors meet the requirements (vendors will need to convert the YAMLs to the format they support). Vendors are required to open a PR containing an instruction doc for reproducing the test result, and the community reviews and approves the PR (Borrowed the idea from [Kubernetes Conformance Test][Kubernetes Conformance Test]). This is a one-time test, but the test result MUST be reproducible when the Conformance claim is valid.
+
+### Q&A
+#### 1. Do field requirements mean the same for Tekton users and vendors?
+
+The short answer is no.
+
+"REQUIRED" in the conformance policy context is, **vendors** have to support this field, while for **users** when submitting Tekton Primitives, those fields are not necessarily required. **Required fields in Tekton APIs must be all included in the required fields of Tekton Conformance APIs.**
+
+For instance, say `TaskRun.Spec.Param` is required in the **Tekton Conformance APIs**, while it is an optional field in the **Tekton APIs**:
+* Tekton **users** are **NOT** required to specify the `Param` field when defining a TaskRun.
+* Tekton **vendors MUST** support that field so that Tekton ***users*** are able to use it.
+
+**Required fields in Tekton APIs must be all included in the required fields of Tekton Conformance APIs.**
+
+If a field is required in Tekton APIs, say `taskSpec.steps`, but is not required in Conformant APIs, vendors have the freedom to not support that field.
```

Comment:
these two sentences seem to conflict with each other

Reply:
I rephrased a bit to emphasizing they are different between **Tekton APIs** (for users) and **Tekton Conformance APIs** (for vendors), hope I successfully expressed myself clearly!

Reply:
I'm still confused; you're saying that required fields in tekton apis must be required in the conformance apis, and give the example of a field (steps) which is required in the tekton apis but not required in the conformance apis.

Reply:
Yes yes, that's what I mean, because in this way, if users are not able to submit the previously conformant YAML to the new vendor Service without the support of the deprecated field.

Reply:
Is the example with `taskspec.steps` an example of something that is not allowed? or an example of something that would require a policy bump?

Reply:
Yes that's right, clarified that!

Reply:
This is something won't ever happen (a field is required in the Tekton APIs but not required in the Conformance API). updated the doc to make it clearer!